File: solutions/swift/kindergarten-garden/1/Sources/KindergartenGarden/KindergartenGarden.swift
enum Plants {
  case clover, grass, radishes, violets
  init(_ ch: Character) {
    switch ch.lowercased() {
    case "c": self = .clover
    case "g": self = .grass
    case "r": self = .radishes
    case "v": self = .violets
    default: fatalError("Unknown plant '\(ch)'")
    }
  }
}

class Garden {
  let rows: [[Character]]

  init(_ diagram: String) {
    rows = diagram.split(separator: "\n").map(Array.init)
    precondition(rows.count == 2)
  }

  func plantsForChild(_ child: String) -> [Plants] {
    let index = Int((child.first!.asciiValue!) - Character("A").asciiValue!)
    return [
      Plants(rows[0][2 * index]),
      Plants(rows[0][2 * index + 1]),
      Plants(rows[1][2 * index]),
      Plants(rows[1][2 * index + 1]),
    ]
  }
}

File: solutions/lua/kindergarten-garden/1/kindergarten-garden.lua
return function(s)
	local all_plants = { C = "clover", G = "grass", R = "radishes", V = "violets" }
	local row1, row2 = s:match("(%w+)\n(%w+)")
	return {
		plants = function(child)
			local index = 2 * (child:byte() - 65) + 1
			return {
				all_plants[row1:sub(index, index)],
				all_plants[row1:sub(index + 1, index + 1)],
				all_plants[row2:sub(index, index)],
				all_plants[row2:sub(index + 1, index + 1)],
			}
		end,
	}
end
